docs/content/docs/data-structures/counter.md

```diff
@@ -1,5 +1,196 @@
 ---
 title: SharedCounter
-menuPosition: 9
-status: unwritten
+menuPosition: 5
 ---
+
+## Introduction
+
+The `SharedCounter` distributed data structure (DDS) is used to store an integer counter value that can be modified collaboratively.
+The data structure affords incrementing and decrementing the shared value via its `increment` method. Decrements are done by providing a negative value.
+
+The `SharedCounter` is a specialized [optimistic DDS]({{< relref "dds.md#optimistic-data-structures" >}}).
+It operates on communicated _deltas_ (amounts by which the shared value should be incremented or decremented), rather than direct changes to the shared value.
+In this way, it avoids the pitfalls of DDSes with simpler merge strategies, in which one user's edit may clobber another's (see [below](#why-a-specialized-dds)).
+
+Note that the `SharedCounter` only operates on integer values.
+
+### Why a specialized DDS?
+
+You may be asking yourself, why not just store the shared integer value directly in another DDS like a [SharedMap][]?
+Why incur the overhead of another runtime type?
+
+The key to the answer here is that DDSes with simpler merge strategies (like `SharedMap`) take a somewhat brute-force approach to merging concurrent edits.
+For a semantic data type like a counter, this can result in undesirable behavior.
+
+#### SharedMap Example
+
+Let's illustrate the issue with an example.
+
+Here, we will use a [SharedMap][] to store our shared integer value.
+For simplicity, it will be stored under a static key: `counter-key`.
+
+Let's say that two users are collaborating on an app with a counter widget.
+We will refer to these users as User A and User B.
+The current value of that widget is 42.
+
+What if User A and User B both simultaneously press the `+` button in their UI to increment the current value by 1?
+Behind the scenes, this increment is implemented by writing the updated value to the map.
+Each edit is then sequenced and broadcast to other collaborators.
+
+For the purpose of this example, we are ignoring the fact that two operations will never really occur at exactly the same time.
+This is intentional.
+Since increments and decrements are [commutative](https://en.wikipedia.org/wiki/Commutative_property), it shouldn't matter in what order two nearly concurrent increment operations occur; the result _should_ be the same.
+
+Here, we would expect that, after both users have processed all incoming edits, the resulting value of the counter would be 44.
+Each user pressed the `+` button once.
+42 + 1 + 1 = **44**, right?
+Again, note that we are ignoring which `+ 1` came from which user, as the order should not matter.
+
+Unfortunately, because `SharedMap` employs a _last-write-wins_ merge strategy, if User A and User B make their edits at the same time, one of the two updates will be sequnced after the other, and that value will win.
```

Comment:
**around** the same time?

Reply:
@ssimic2 I made a note a couple paragraphs earlier about handwaving the simultaneity issue. Do you think that's sufficient? Or do you think it would be worth updating phrasing to be more explicitly approximate?

> For the purpose of this example, we are ignoring the fact that two operations will never really occur at exactly the same time.
> This is intentional.

Reply:
I ended up completely re-working this example. It is much more explicit about the ordering of things, and uses looser language around timing.